### ISSUE TITLE
WriteLoop issues

### ISSUE THREAD

**Issue by backkem** [pion/sctp#42]
### Your environment.
  - Version: 1.6.0

### Summary
Running SCTP v1.6.0 as part of the libp2p / go-libp2p-webrtc-direct CI ([logs](https://travis-ci.org/libp2p/go-libp2p-webrtc-direct/builds/524485205)) still seems to cause some problems that are not limited to ICE:
- There seems to be a problem with shutdown where SCTP keeps trying to write even thought the DTLS connection is already closed: [log](https://api.travis-ci.org/v3/job/524451909/log.txt) (this has ICE rolled back as well). I tried to end the writeLoop when we can't write to DTLS in [pion/sctp/writeloop](https://github.com/pion/sctp/tree/writeloop) but this doesn't fix the problem entirely:  [log](https://api.travis-ci.org/v3/job/524485206/log.txt)
- `SubtestPingPong` is failing as well (disabled in the CI logs linked above). These tests expect the Read & Write methods to operate with io.ReadWriter like semantics, E.g. return io.EOF, etc. My feeling is that these semantics may have changed in v1.6.0. This needs more digging, thought.

cc @enobufs @Sean-Der

**Comment by enobufs** [comment]
Thanks @backkem, great catch! Looks like the goroutine for T3-rtx timer is still active.  (probably other timers, too) These are cleaned up when association.Close() with the current code, and it probably needs to be cleaned up when writeLoop ends.

**Comment by enobufs** [comment]
Hmm, I see this also in your log:
```
sctp DEBUG: 13:53:52.877254 association.go:499: retransmitting 900 bytes
```
The writeLoop has not ended with some node... I guess I need to improve the log message for dig deeper.

**Comment by enobufs** [comment]
I confirmed SubtestPingPong test fails on my mac as well.
Looking at the log (the improved one I just pushed to writeLoop branch), there are two nodes (associations).  One end terminates (I can see "writeLoop ended" and "closing association.."), but the other keeps trying to retransmit packets for about 45 seconds, then finally sees an write error on netConn. (I do not see, so I guess it never receives association.Close() and write succeeds during the 45 seconds (netConn is still open)...

**Comment by enobufs** [comment]
@bacckem
With the test, SubtestStress50Conn10Stream50Msg, I was able to see some nodes firing times even though writeLoop exited.

I added a code to clean up timers when exiting writeLoop - I pushed the change to the branch.  I ran the same test, SubtestStress50Conn10Stream50Msg, I no longer see a node firing rtx timeout after writeLoop exits, so it looks a bit clearner on the log but, I still see some nodes trying to retransmit packets on T3-rtx timeout but from the log. Those nodes never see association.Close() nor write error on netConn, so again, dtls is still open and writeLoop is (legitimately) running, I think.
